Documentation/Design/Design Doc Template.md

```diff
@@ -0,0 +1,151 @@
+# Feature Name
+
+## Feature Overview
+
+A description of the feature.
+
+
+## Commands
+
+### `command 1`
+
+`command 1 syntax`
+
+A brief description of the command, including any optional parameters or parameters with default values.
+
+### `command 2`
+
+`command 2 syntax`
+
+A brief description of the command, including any optional parameters or parameters with default values.
+
+## Usage
+
+```gls
+command 1 examples
+```
+
+```gls
+command 2 examples
+```
+
+### CSharp
+
+```csharp
+command 1 C# output
+```
+
+```csharp
+command 2 C# output
+```
+
+### Java
+
+```java
+command 1 java output
+```
+
+```java
+command 2 java output
+```
+
+### Python
+
+```python
+command 1 python output
+```
+
+```python
+command 2 python output
+```
+
+### Ruby
+
+```ruby
+command 1 ruby output
+```
+
+```ruby
+command 1 ruby output
+```
+
+### TypeScript
+
+```typescript
+command 1 TypeScript output
+```
+
+```typescript
+command 1 TypeScript output
+```
+
+
+## Implementation
+
+### Properties
+
+<table>
+    <thead>
+        <th>Name</th>
+        <th>Type</th>
+        <th>Description</th>
+    </thead>
+    <tbody>
+        <tr>
+            <td>Property1</td>
+            <td><code>type</code></td>
+            <td>Description of property1.</td>
+        </tr>
+        <tr>
+            <td>Property2</td>
+            <td><code>type</code></td>
+            <td>Description of property2.</td>
+        </tr>
+    </tbody>
+</table>
+
+### Language Values
+
+<table>
+    <thead>
+        <th>Language</th>
+        <th>Property1</th>
+        <th>Property2</th>
+    </thead>
+    <tbody>
+        <tr>
+            <th>CSharp</th>
+            <td><code>C# Property1 value</code></td>
+            <td><code>C# Property2 value</code></td>
+        </tr>
+        <tr>
+            <th>Java</th>
+            <td><code>Java Property1 value</code></td>
+            <td><code>Java Property2 value</code></td>
+        </tr>
+        <tr>
+            <th>Ruby</th>
+            <td><code>Ruby Property1 value</code></td>
+            <td><code>Ruby Property2 value</code></td>
+        </tr>
+        <tr>
+            <th>Python</th>
+            <td><code>Python Property1 value</code></td>
+            <td><code>Python Property2 value</code></td>
+        </tr>
+        <tr>
+            <th>TypeScript</th>
+            <td><code>TypeScript Property1 value</code></td>
+            <td><code>TypeScript Property2 value</code></td>
+        </tr>
+    </tbody>
+</table>
+
+### Tests
+
+Any relevant notes about test coverage. 
+
+### Errata
+
+* Extra stuff about languages that do or do not support certain features
+* Other items that are out of scope
```

Comment:
They're bullet points, so probably not required, but we can have them. I'm not picky either way.